src/SlashCommands.js

```diff
@@ -34,6 +34,26 @@ import WidgetUtils from "./utils/WidgetUtils";
 import {textToHtmlRainbow} from "./utils/colour";
 import Promise from "bluebird";
 
+const singleMxcUpload = async () => {
+    return new Promise((resolve) => {
+        const fileSelector = document.createElement('input');
+        fileSelector.setAttribute('type', 'file');
+        fileSelector.onchange = (ev) => {
+            const file = ev.target.files[0];
+
+            const UploadConfirmDialog = sdk.getComponent("dialogs.UploadConfirmDialog");
+            Modal.createTrackedDialog('Upload Files confirmation', '', UploadConfirmDialog, {
+                file,
+                onFinished: (shouldContinue) => {
+                    if (shouldContinue) resolve(MatrixClientPeg.get().uploadContent(file));
```

Comment:
hmmm... something I didn't notice before was that we weren't rejecting the promise otherwise. This ends up leaving things open for far longer than their lifespan - can we throw an `else reject()` on here (or `resolve(null)`)?

Reply:
Fixed, reject did not play nicely without a catch, caused an ambiguous error popup, so `resolve(null)` and a null-guard it is